assets/app/view/game/game_meta.rb

```diff
@@ -3,7 +3,13 @@
 module View
   module Game
     class GameMeta < Snabberb::Component
-      needs :game
+      needs :game, default: nil
+      needs :game_class, default: nil
+
+      def initialize(*)
+        super
+        @game_class = @game.class unless defined? @game_class
```

Comment:
what's going on here... there's got to be a better way,

@michaeljb suggestions?

Reply:
`View::Game::GameMeta` expects `game` to be an `Engine::Game::…::Game` object, but in `View::CreateGame` we have an `Engine::Game::…::Meta` class, not an instantiated object.  Since `View::Game::GameMeta` mostly accesses class variables that are common between the `Game` and the `Meta` classes, I wanted to be able to be able to pass it a class rather than an instance.

The obvious thing I tried was:

```ruby
needs :game, default: nil
needs :game_class, default: @game.class
```

However, this doesn’t work because `@game` doesn’t get set early enough.

The other obvious thing I tried was to instantiate the `Meta` class before passing it to `View::Game::GameMeta`, but this throws, e.g., ``undefined method `new' for Engine::Game::G1817::Meta``.  I admit I don’t understand why that is, and whether it would be reasonable to “fix” it.

Reply:
`Meta` is a module, not a class, so that's why it cannot be instantiated with `new`.

Looks like the only thing in `GameMeta` that needs an actual `Game` instance is the optional rules bit--it wants to show which optional rules were used in creating the current game.

Probably can add a little bit of logic in `GameMeta#render` to determine if `@game` is a `Game` instance or a `Meta` module, I'll play around with it a bit.

Reply:
@mavit I've made a PR onto your branch, if you merge it, this PR will be updated automatically

https://github.com/mavit/18xx/pull/1

This adds `.meta` and `.game_instance?` methods to `Game` classes, their instances, and `Meta` modules, so the `Meta` is always easy to reach now.

![Screenshot 2021-03-15 092032](https://user-images.githubusercontent.com/1045173/111177399-cb15ce00-856f-11eb-9d29-7899e9db4965.png)